mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java

```diff
@@ -703,24 +726,36 @@ public void onSuccess(Call call, int statusCode, Headers headers, byte[] respons
     }
 
     private void showSitemapSelectionDialog(final List<OpenHABSitemap> sitemapList) {
+        // We need a lock here so the dialog is only shown once at a time
+        if(showSitemapSelectionDialogLock.isLocked()) {
```

Comment:
Why do we need to use threading locks here?  What is calling this twice?  

Also I'm not sure if this is the correct pattern for using a Reentrant Lock, for example in this line I think you want to attain the lock only if its not currently locked, so this would be the "tryLock" method, checking it first then trying to lock would leave a chance another thread could lock in between those calls.  Also locks need to be released in a finally block.

Reply:
Indeed; also as this is changing UI, it can be called by the UI thread only by definition anyway.
I think the idea is to avoid showing the dialog twice; the correct solution for that would be keeping a reference to the dialog and check whether isShowing() returns true.

Reply:
After the intro is finished, onResume is called and that calls resetStateFragmentAfterResume => mOpenHABTracker.start() =>openHABTracked() => onOpenHABTracked() =>selectSitemap() =>showSitemapSelectionDialog()

A better fix would be to wait in the main activity until the intro is finished

Reply:
'Wait in the main activity'? What do you mean?

Reply:
Nevermind. The issue is that the main activity is started and after the intro it's resumed and both action display the select sitemap dialog.

Reply:
@maniac103 How can I check if it's still shown? AlertDialog.Builder doesn't seem to have a function for that.